include/hip/nvcc_detail/hip_runtime_api.h

```diff
@@ -244,6 +244,13 @@ typedef struct cudaResourceViewDesc hipResourceViewDesc;
 #define HIP_FUNC_ATTRIBUTE_PREFERRED_SHARED_MEMORY_CARVEOUT CU_FUNC_ATTRIBUTE_PREFERRED_SHARED_MEMORY_CARVEOUT
 #define HIP_FUNC_ATTRIBUTE_MAX CU_FUNC_ATTRIBUTE_MAX
 
+#if CUDA_VERSION >= 9020
+#define __shfl(var, srcLane, width)      __shfl_sync(0xffffffff, var, srcLane, width)
+#define __shfl_up(var, delta, width)     __shfl_up_sync(0xffffffff, var, delta, width)
+#define __shfl_down(var, delta, width)   __shfl_down_sync(0xffffffff, var, delta, width)
+#define __shfl_xor(var, laneMask, width) __shfl_xor_sync(0xffffffff, var, laneMask, width)
+#endif // CUDA_VERSION >= 9020
```

Comment:
`__shfl_*_sync` are supported since CUDA 9.0: `CUDA_VERSION >= 9000`

Reply:
In addition, these should be static inline functions or variadic macros such that they can use a default argument of `width=warpSize`.
This is currently causing the failure in the hipShflTests.cpp, which uses the two argument form:

```
[cuda_10_x] /var/jenkins/workspace/Developer-Tools_HIP_PR-2074-NYMVQWTWJHPYBXD44TIHEY7TVICCRCNNYPFMV5FUBTNWQ4XRLQYA/src/hip/tests/src/kernel/hipShflTests.cpp:46:87: error: macro "__shfl" requires 3 arguments, but only 2 given
[cuda_10_x]          for (int j = 0; j < width; j++) out[i * width + j] = **__shfl(val, j * width + i);**
[cuda_10_x] 
```

Reply:
Added separate defines for  with=wrapSize